### PR TITLE
Index page should honor configured HTTP path prefix.

### LINKED ISSUE #3118: http_path_prefix not always respected
We are running cortex in a cluster form. so we are starting several containers with the target config set.

When starting a distributor with: `http_path_prefix: "/distributor"` in the config, the index page is loaded from the domain.com/distributor page in the browser.
when clicking on `usage statistics` it redirects to `/distributor/all_user_statistics`
but clicking any other link will redirect to their corresponding page on `domain.com` instead of `domain.com/distributor`

to me it seems like not all the urls respect the `http_path_prefix` from the config

using cortex:1.3.0

### PR DESCRIPTION
**Pull request by pstibrany**
**What this PR does**: This PR fixes index page, and adds configured HTTP path prefix to all internal links it renders.

**Which issue(s) this PR fixes**:
Fixes #3118.

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/api/handlers.go [hunk 1]
```diff
@@ -21,29 +23,39 @@ const indexPageContent = `
...
 
-func indexHandler(w http.ResponseWriter, _ *http.Request) {
-	if _, err := w.Write([]byte(indexPageContent)); err != nil {
-		http.Error(w, err.Error(), http.StatusInternalServerError)
-		return
+type templateInp struct {
+	Prefix string
+}
+
+func (i templateInp) Join(p string) string {
```
Comment (sandeepsukhani) [inline pkg/api/handlers.go:49]: Should we call this `PrefixedPath` or something?

**Review by sandeepsukhani — APPROVED** [review]
Just a minor suggestion otherwise it LGTM!

**Review by jtlisi — APPROVED** [review]
LGTM, good change

File: pkg/api/handlers.go (hunk 1 above)
Comment (pstibrany) [reply pkg/api/handlers.go]: What about `AddPathPrefix`?

Comment (sandeepsukhani) [reply pkg/api/handlers.go]: Sounds good!

**Pull request merged**